test/Microsoft.SqlTools.ServiceLayer.UnitTests/Profiler/ProfilerServiceTests.cs

```diff
@@ -104,23 +107,78 @@ public class ProfilerServiceTests
+            profilerService.SessionMonitor.StartMonitoringSession(testUri, mockSession.Object);
 
             await profilerService.HandleStopProfilingRequest(requestParams, requestContext.Object);
 
             requestContext.VerifyAll();
 
-            // check that session was succesfully stopped and drop was called
+            // check that session was succesfully stopped and stop was called
             Assert.True(success);
             Assert.True(stopped);
 
             // should not be able to remove the session, it should already be gone
             ProfilerSession ps;
             Assert.False(profilerService.SessionMonitor.StopMonitoringSession(testUri, out ps));
         }
+
+        /// <summary>
+        /// Test pausing then resuming a session
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        public async Task TestPauseProfilingRequest()
+        {
+            bool success = false;
+            string testUri = "test_session";
+
+            // capture stopping results
+            var requestContext = new Mock<RequestContext<PauseProfilingResult>>();
+            requestContext.Setup(rc => rc.SendResult(It.IsAny<PauseProfilingResult>()))
+                .Returns<PauseProfilingResult>((result) =>
+                {
+                    success = result.Succeeded;
+                    return Task.FromResult(0);
+                });
+
+            var sessionListener = new TestSessionListener();
+            var profilerService = new ProfilerService();
+            profilerService.SessionMonitor.AddSessionListener(sessionListener);
+            profilerService.ConnectionServiceInstance = TestObjects.GetTestConnectionService();
+            ConnectionInfo connectionInfo = TestObjects.GetTestConnectionInfo();
+            profilerService.ConnectionServiceInstance.OwnerToConnectionMap.Add(testUri, connectionInfo);
+
+            var requestParams = new PauseProfilingParams();
+            requestParams.OwnerUri = testUri;
+
+            profilerService.SessionMonitor.StartMonitoringSession(testUri, new TestXEventSession1());
+
+            // wait a bit for profile sessions to be polled
+            Thread.Sleep(1500);
+
+            Assert.Equal(sessionListener.PreviousSessionId, testUri);
+
+            // pause viewer
+            await profilerService.HandlePauseProfilingRequest(requestParams, requestContext.Object);
+            Assert.True(success);
+
+            sessionListener.PreviousSessionId = "";
+            success = false;
+
+            Thread.Sleep(1500);
+
+            // check that session wasn't polled, and no events were sent to listener
+            Assert.Equal(sessionListener.PreviousSessionId, "");
+
+            // unpause viewer
+            await profilerService.HandlePauseProfilingRequest(requestParams, requestContext.Object);
+            Assert.True(success);
+
+            Thread.Sleep(1500);
```

Comment:
sleeping this long in a test, especially multiple times, isn't great. I know a bunch of our tests already call sleep a bit but we should try to avoid it in new code. I don't think there's any risk of flakiness here like there is in a lot of our other tests where we call sleep because it's only waiting for polling to occur (right?) but is there any way to avoid sleeping here? For example maybe some way to poll manually when testing?